OrbitCore/OrbitFunction.cpp

```diff
@@ -160,6 +158,40 @@ const char* Function::GetCallingConventionString() {
   return kCallingConventions[calling_convention_];
 }
 
+const absl::flat_hash_map<const char*, Function::OrbitType>&
+Function::GetFunctionNameToOrbitTypeMap() {
+  static absl::flat_hash_map<const char*, OrbitType> function_name_to_type_map{
+      {"Start(", ORBIT_TIMER_START},
+      {"Stop(", ORBIT_TIMER_STOP},
+      {"StartAsync(", ORBIT_TIMER_START_ASYNC},
+      {"StopAsync(", ORBIT_TIMER_STOP_ASYNC},
+      {"TrackInt(", ORBIT_TRACK_INT},
+      {"TrackInt64(", ORBIT_TRACK_INT_64},
+      {"TrackUint(", ORBIT_TRACK_UINT},
+      {"TrackUint64(", ORBIT_TRACK_UINT_64},
+      {"TrackFloat(", ORBIT_TRACK_FLOAT},
+      {"TrackDouble(", ORBIT_TRACK_DOUBLE},
+      {"TrackFloatAsInt(", ORBIT_TRACK_FLOAT_AS_INT},
+      {"TrackDoubleAsInt64(", ORBIT_TRACK_DOUBLE_AS_INT_64},
+  };
+  return function_name_to_type_map;
+}
+
+// Detect Orbit API functions by looking for special function names part of the
+// orbit namespace. If there is a match, set the corresponding function type.
+bool Function::SetOrbitTypeFromName() {
+  const std::string& name = PrettyName();
+  if (absl::StartsWith(name, "orbit::")) {
+    for (auto& pair : GetFunctionNameToOrbitTypeMap()) {
+      if (absl::StrContains(name, pair.first)) {
```

Comment:
Considering the case when we profile Orbit itself: If I read this right, then if I have a function in orbit::something::etc and it contains any of the Api function names  (like "StartCapture"), it will be considered a call to our instrumentation API. 

If this is correct, and even though this only affects a rare case: Should we have clear namespace rules that would prevent this? For example, orbit_api::Start for the instrumentation methods. (I looked through the code and our use of namespaces is all over the place in terms of style, so simplifying this would also help with consistency.)

Reply:
Yes, you're right, using "orbit_api" is a good idea, I'll do that, thanks.